app/src/main/java/com/m/androidNativeApp/MainActivity.java

```diff
@@ -178,51 +199,73 @@ public void addTaskActivity(View view) {
 
     public void getTasks() {
 
+        ResponseFetcher onlineResponse = ApolloResponseFetchers.NETWORK_ONLY;
+
+        if (!isOnline()) {
+            onlineResponse = ApolloResponseFetchers.CACHE_ONLY;
+        }
+
         AllTasksQuery tasksQuery = AllTasksQuery
                 .builder()
                 .build();
 
-        client.query(tasksQuery).enqueue(new ApolloCall.Callback<AllTasksQuery.Data>() {
-            @Override
-            public void onResponse(@NotNull Response<AllTasksQuery.Data> response) {
-                final int dataLength = response.data().allTasks().size();
-
-                for (int i = 0; i < dataLength; i++) {
-                    TaskFields dataReceived = response.data().allTasks().get(i).fragments().taskFields();
-                    taskTitle = dataReceived.title();
-                    taskDescription = dataReceived.description();
-                    taskId = dataReceived.id();
-                    itemList.add(new Item(taskTitle, taskDescription, taskId));
-                }
-                runOnUiThread(() -> itemAdapter.notifyDataSetChanged());
-            }
-
-            @Override
-            public void onFailure(@NotNull ApolloException e) {
-                if (e.getMessage().equals("HTTP 403 Forbidden")) {
-                    reAuthorise();
-                }
-            }
-        });
+        client.query(tasksQuery)
+                .responseFetcher(onlineResponse)
+                .enqueue(new ApolloCall.Callback<AllTasksQuery.Data>() {
+                    @Override
+                    public void onResponse(@NotNull Response<AllTasksQuery.Data> response) {
+                        final int dataLength = response.data().allTasks().size();
+                        for (int i = 0; i < dataLength; i++) {
+                            TaskFields dataReceived = response.data().allTasks().get(i).fragments().taskFields();
+                            taskTitle = dataReceived.title();
+                            taskDescription = dataReceived.description();
+                            taskId = dataReceived.id();
+                            itemList.add(new Item(taskTitle, taskDescription, taskId));
+                        }
+
+
+                        client.apolloStore().read(tasksQuery);
```

Comment:
what does this call do?

Reply:
Sorry, I was testing something and it should not be here.